### PR TITLE
Exposing a builder context as part of the startup

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

File: src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+using System;
+using System.IO;
+
+namespace Microsoft.Azure.Functions.Extensions.DependencyInjection
+{
+    public abstract class FunctionHostBuilderContext
+    {
+        private string _rootPath;
+
+        public string RootPath
```
Comment (brettsam) [inline src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs:13]: nit: I was calling this `ApplicationRootPath` but I also know that it's called `ContentRootPath` in [IHostingEnvironment](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.ihostingenvironment.contentrootpath?view=aspnetcore-3.1#Microsoft_AspNetCore_Hosting_IHostingEnvironment_ContentRootPath). Should we pick one of those to be a little more specific?

File: src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+                    if (home != null)
+                    {
+                        _rootPath = Path.Combine(home, "site", "wwwroot");
+                    }
+                    else
+                    {
+                        _rootPath = Environment.GetEnvironmentVariable("AzureWebJobsScriptRoot");
+                    }
+                }
```
Comment (brettsam) [inline src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs:29]: Is this the same pattern as on Linux?

File: src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs (hunk 1 above)
Comment (fabiocav) [reply src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs:13]: I was going with the `FunctionsContext.RootPath`approach, but `ApplicationRootPath` makes sense.

File: src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs (hunk 2 above)
Comment (fabiocav) [reply src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs:29]: Yes, it is

Comment (fabiocav) [reply src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs:29]: Pretty sure I've validated, but it has been a while, @ahmelsayed can you confirm?

Comment (ahmelsayed) [reply src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs:29]: Yes, that pattern is correct. $HOME on linux workers is /home, and the app content is in /home/site/wwwroot/

File: src/Extensions/DependencyInjection/FunctionsHostBuilder.cs [hunk 3]
```diff
@@ -3,18 +3,26 @@
...
 {
-    internal class FunctionsHostBuilder : IFunctionsHostBuilder
+    internal class FunctionsHostBuilder : IFunctionsHostBuilder, IFunctionsHostBuilderExt
     {
+        public FunctionsHostBuilder()
+        {
+            Context = new DefaultFunctionHostBuilderContext();
+        }
+
         public FunctionsHostBuilder(IServiceCollection services)
```
Comment (brettsam) [inline src/Extensions/DependencyInjection/FunctionsHostBuilder.cs:19]: Call `base()`?

File: src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs (hunk 2 above)
Comment (brettsam) [reply src/Extensions/DependencyInjection/FunctionHostBuilderContext.cs:29]: After my PR here (https://github.com/Azure/azure-functions-host/pull/5991) -- I'll be able to pass this in directly, taking it from the options in the WebHost. So maybe we don't need this logic at all once I follow-up with that wire-up.

**Pull request merged**